File: typescript-examples/quick-recipes/api/upload-to-s3.ts
import { BrowserContext, Page } from "playwright";
import { saveFileToS3 } from "@intuned/browser";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  // Navigate to the file list
  await page.goto("https://sandbox.intuned.dev/pdfs");

  // Locate the download trigger for the first row
  const downloadLocator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']");

  // Download and upload to S3 in one step
  // https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/helpers/functions/saveFileToS3
  const uploadedFile = await saveFileToS3({
    page,
    trigger: downloadLocator,
  });

  // Return signed URL for access
  const signedUrl = await uploadedFile.getSignedUrl();
  console.log(`Download uploaded file at: ${signedUrl}`);
  return signedUrl;
}

File: python-examples/quick-recipes/api/upload-to-s3.py
from typing import TypedDict

from intuned_browser import save_file_to_s3
from playwright.async_api import Page


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params | None = None, **_kwargs):
    # Navigate to the file list
    await page.goto("https://sandbox.intuned.dev/pdfs")

    # Locate the download trigger for the first row
    download_locator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']")

    # Download and upload to S3 in one step
    # https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/helpers/functions/save_file_to_s3
    uploaded_file = await save_file_to_s3(
        page=page,
        trigger=download_locator,
    )

    # Return signed URL for access
    signed_url = await uploaded_file.get_signed_url()
    print(f"Download uploaded file at: {signed_url}")
    return signed_url
